service/src/main/java/greencity/service/HabitAssignServiceImpl.java

```diff
@@ -39,101 +37,178 @@
      * {@inheritDoc}
      */
     @Override
-    public HabitAssignDto getById(Long habitAssignId) {
-        return modelMapper.map(habitAssignRepo.findById(habitAssignId)
-            .orElseThrow(() -> new NotFoundException(ErrorMessage.HABIT_ASSIGN_NOT_FOUND_BY_ID + habitAssignId)),
-            HabitAssignDto.class);
+    public HabitAssignDto getById(Long habitAssignId, String language) {
+        HabitAssign habitAssign = habitAssignRepo.findById(habitAssignId)
+            .orElseThrow(() -> new NotFoundException(ErrorMessage.HABIT_ASSIGN_NOT_FOUND_BY_ID + habitAssignId));
+
+        HabitDto habitDto = habitService.getByIdAndLanguageCode(habitAssign.getHabit().getId(), language);
+        HabitAssignDto habitAssignDto = modelMapper.map(habitAssign, HabitAssignDto.class);
+        habitAssignDto.setHabit(habitDto);
+        return habitAssignDto;
     }
 
     /**
      * {@inheritDoc}
      */
     @Transactional
     @Override
-    public HabitAssignDto assignHabitForUser(Long habitId, UserVO userVO) {
+    public HabitAssignManagementDto assignDefaultHabitForUser(Long habitId, UserVO userVO) {
+        User user = modelMapper.map(userVO, User.class);
+        Habit habit = validateHabitForAssign(habitId, user);
+
+        HabitAssign habitAssign = buildHabitAssign(habit, user);
+        enhanceAssignWithDefaultProperties(habitAssign);
+
+        habitStatusService.saveStatusByHabitAssign(modelMapper.map(habitAssign, HabitAssignVO.class));
+        return modelMapper.map(habitAssign, HabitAssignManagementDto.class);
+    }
+
+    /**
+     * Method updates {@link HabitAssign} with default properties.
+     *
+     * @param habitAssign {@link HabitAssign} instance.
+     */
+    private void enhanceAssignWithDefaultProperties(HabitAssign habitAssign) {
+        habitAssign.setDuration(habitAssign.getHabit().getDefaultDuration());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Transactional
+    @Override
+    public HabitAssignManagementDto assignCustomHabitForUser(Long habitId, UserVO userVO,
+        HabitAssignPropertiesDto habitAssignPropertiesDto) {
+        User user = modelMapper.map(userVO, User.class);
+        Habit habit = validateHabitForAssign(habitId, user);
```

Comment:
validation method should be void or return some exceptions. Looks strange that it returns habit. Please, refactor it

Reply:
Changes done.